### ISSUE TITLE
[mqtt.homeassistant] Rollershutter from z-wave-js-ui work only partly

### ISSUE THREAD

**Issue by chaosjug** [openhab/openhab-addons#15845]
I have a Fibaro FGR222 connected to a z-wave stick which is managed by z-wave-js. In Openhab I try to use homeassistant autodiscovery to add the device to openHAB.
The rollershutter detected and shows up in the inbox as expected. But when I add the thing, no channels are there. At some point they appear, not sure exactly when but restarting z-wave-js seems to help.
The state of the thing is "unknown" until the channels show up.
This shows up in the log when adding the thing:
```
2023-11-04 23:55:06.519 [WARN ] [re.thing.internal.ThingFactoryHelper] - Could not create channel 'zwavejs2mqtt_5F0xf19aa6c2_5F3_2D49_2D0_2DPower#sensor', because channel type 'mqtt:zwavejs2mqtt_5F0xf19aa6c2_5F3_2D49_2D0_2DPower_sensor' could not be found.
2023-11-04 23:55:06.520 [WARN ] [re.thing.internal.ThingFactoryHelper] - Could not create channel 'zwavejs2mqtt_5F0xf19aa6c2_5F3_2D37_2D0_2DcurrentValue#switch', because channel type 'mqtt:zwavejs2mqtt_5F0xf19aa6c2_5F3_2D37_2D0_2DcurrentValue_switch' could not be found.
2023-11-04 23:55:06.521 [WARN ] [re.thing.internal.ThingFactoryHelper] - Could not create channel 'zwavejs2mqtt_5F0xf19aa6c2_5F3_2D50_2D0_2Dvalue_2D66049#sensor', because channel type 'mqtt:zwavejs2mqtt_5F0xf19aa6c2_5F3_2D50_2D0_2Dvalue_2D66049_sensor' could not be found.
2023-11-04 23:55:06.521 [WARN ] [re.thing.internal.ThingFactoryHelper] - Could not create channel 'zwavejs2mqtt_5F0xf19aa6c2_5FNode3_5Fposition#cover', because channel type 'mqtt:zwavejs2mqtt_5F0xf19aa6c2_5FNode3_5Fposition_cover' could not be found.
2023-11-04 23:55:06.521 [WARN ] [re.thing.internal.ThingFactoryHelper] - Could not create channel 'zwavejs2mqtt_5F0xf19aa6c2_5F3_2D50_2D0_2Dvalue_2D65537#sensor', because channel type 'mqtt:zwavejs2mqtt_5F0xf19aa6c2_5F3_2D50_2D0_2Dvalue_2D65537_sensor' could not be found.
2023-11-04 23:55:06.521 [WARN ] [re.thing.internal.ThingFactoryHelper] - Could not create channels for channel group 'zwavejs2mqtt_5F0xf19aa6c2_5F3_2D49_2D0_2DPower' for thing type 'mqtt:homeassistant_zwavejs2mqtt_5F0xf19aa6c2_5Fnode3:474fb7f2f7:zwavejs2mqtt_5F0xf19aa6c2_5Fnode3', because channel group type 'mqtt:zwavejs2mqtt_5F0xf19aa6c2_5F3_2D49_2D0_2DPower' could not be found.
2023-11-04 23:55:06.521 [WARN ] [re.thing.internal.ThingFactoryHelper] - Could not create channels for channel group 'zwavejs2mqtt_5F0xf19aa6c2_5F3_2D37_2D0_2DcurrentValue' for thing type 'mqtt:homeassistant_zwavejs2mqtt_5F0xf19aa6c2_5Fnode3:474fb7f2f7:zwavejs2mqtt_5F0xf19aa6c2_5Fnode3', because channel group type 'mqtt:zwavejs2mqtt_5F0xf19aa6c2_5F3_2D37_2D0_2DcurrentValue' could not be found.
2023-11-04 23:55:06.521 [WARN ] [re.thing.internal.ThingFactoryHelper] - Could not create channels for channel group 'zwavejs2mqtt_5F0xf19aa6c2_5F3_2D50_2D0_2Dvalue_2D66049' for thing type 'mqtt:homeassistant_zwavejs2mqtt_5F0xf19aa6c2_5Fnode3:474fb7f2f7:zwavejs2mqtt_5F0xf19aa6c2_5Fnode3', because channel group type 'mqtt:zwavejs2mqtt_5F0xf19aa6c2_5F3_2D50_2D0_2Dvalue_2D66049' could not be found.
2023-11-04 23:55:06.521 [WARN ] [re.thing.internal.ThingFactoryHelper] - Could not create channels for channel group 'zwavejs2mqtt_5F0xf19aa6c2_5FNode3_5Fposition' for thing type 'mqtt:homeassistant_zwavejs2mqtt_5F0xf19aa6c2_5Fnode3:474fb7f2f7:zwavejs2mqtt_5F0xf19aa6c2_5Fnode3', because channel group type 'mqtt:zwavejs2mqtt_5F0xf19aa6c2_5FNode3_5Fposition' could not be found.
2023-11-04 23:55:06.521 [WARN ] [re.thing.internal.ThingFactoryHelper] - Could not create channels for channel group 'zwavejs2mqtt_5F0xf19aa6c2_5F3_2D50_2D0_2Dvalue_2D65537' for thing type 'mqtt:homeassistant_zwavejs2mqtt_5F0xf19aa6c2_5Fnode3:474fb7f2f7:zwavejs2mqtt_5F0xf19aa6c2_5Fnode3', because channel group type 'mqtt:zwavejs2mqtt_5F0xf19aa6c2_5F3_2D50_2D0_2Dvalue_2D65537' could not be found.
```

When the channels show up I can add the rollershutter channel and connect it to an item.
I can operate the rollershutter from openHAB. Up/down works as expected but position setting is inverted. This is probably due to the fact that fibaro defines 100 as open but openHAB defines 0 as open. More on that below.
What also does not work, is the position feedback. The item always shows the commanded value, even if stop is pressed before.

The json configuration string received by the component via MQTT:
```
{
"command_topic":"zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/targetValue/set",
"position_topic":"zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/currentValue",
"set_position_topic":"zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/targetValue/set",
"position_template":"{{ (value_json.value / 99 * 100) | round(0) }}",
"position_open":99,
"position_closed":0,
"payload_open":"99",
"payload_close":"0",
"device":{"identifiers":["zwavejs2mqtt_0xffffffff_node3"],"manufacturer":"Fibargroup","model":"Roller Shutter 2 (FGR222)","name":"Schlafzimmer-SchlafzimmerRechts","sw_version":"25.25"},
"name":"Schlafzimmer-SchlafzimmerRechts_position","unique_id":"zwavejs2mqtt_0xffffffff_Node3_position"
}
```
When I command it to go down and the stop, I see this with mosquitto_sub (starting from fully open = 100):
```
zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/targetValue/set 0
zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/targetValue {"time":1699133440905,"value":0}
zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/currentValue {"time":1699133440905,"value":0}
zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/targetValue/set STOP
zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/currentValue {"time":1699133445568,"value":73}
```
And the rollershutter moves down, stops and shows 100 in openHAB.

When I command it to go up and the stop, I see this with mosquitto_sub (starting from 2/3 closed = 33):
```
zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/targetValue/set 99
zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/targetValue {"time":1699135271861,"value":99}
zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/currentValue {"time":1699135271861,"value":99}
zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/targetValue/set STOP
zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/currentValue {"time":1699135274120,"value":36}
```
And the rollershutter moves up, stops and shows 0 in openHAB.

When I command it to go to position 3 (which in openhab means mostly open) and the stop, I see this with mosquitto_sub (starting from half closed = 50):
```
zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/targetValue/set 3
zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/targetValue {"time":1699133763769,"value":3}
zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/currentValue {"time":1699133763769,"value":3}
zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/targetValue/set STOP
zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/currentValue {"time":1699133770272,"value":33}
```
But the rollershutter moves down instead of up, and shows 3 in openHAB.

For the up and down command openHAB uses the "position_open" and "position_closed" from the config, which ensures that those commands work correctly. But fails to use this information to calculate the correct position value it would need to send.

Looking at the position feedback, z-wave-js sends the position, but openHAB does not seem to use it. I'm a bit unsure here because at first I didn't have the JINJA addon installed and then the position was always 100. But it still just sets the state to what was commanded.

## Your Environment
* Version used: openHAB 3.4.5
* debian bookwork with openHAB in a docker

**Comment by apella12** [comment]
This caught my eye because I have been testing zwave-js-ui as an alternative Zwave binding (see [forum](https://community.openhab.org/t/test-zwave-js-ui-in-place-of-zwave-binding/150007)).  What I found was that the homeassistant auto discovery has several limitations.  See [docs](https://www.openhab.org/addons/bindings/mqtt.homeassistant/#homeassistant-mqtt-components-binding). To avoid them I used the generic mqtt binding as a workaround (no discovery, but can customize the channel exactly). Unfortunately I do not have a rollershutter, so cannot provide exact guidance, but a couple of thoughts;
1) What is the item type? Is it mqtt:switch or mqtt:dimmer. I'm thinking switch would be limited to 0/99(100), but dimmer (or maybe mqtt:number) might provide the currentValue with your current setup.
2) If that doesn't work you could try the generic mqtt "thing".  I have dimmers, not blinds (using value only) (code tab);
` - id: Basement_Door_Light_n
    channelTypeUID: mqtt:dimmer
    label: Basement Door Light
    description: null
    configuration:
      commandTopic: zwave2/20/38/0/targetValue/set
      stateTopic: zwave2/20/38/0/currentValue`
38=switch_multilevel

**Comment by chaosjug** [comment]
Regarding 1:
Which item do you mean? The homeassistant thing has several channels and one of those is a rollershutter which I can connect to a rollershutter item. And I can set the position as well, it is just inverted.
Regarding 2:
That is what I'm using at the moment, but auto discovery is so much better as it saves a lot of time and make it more accessible to people which do not have the time/knowledge to configure everything by hand.
Just for the record, this is what I currently use:
```
channels:
  - id: SchlafzimmerRollladenRechtsPosition
    channelTypeUID: mqtt:rollershutter
    label: Position
    description: ""
    configuration:
      commandTopic: zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/targetValue/set
      transformationPatternOut: JINJA:{{value if value =="STOP" else (100-value)*99//100}}
      stateTopic: zwave/Schlafzimmer/SchlafzimmerRechts/switch_multilevel/endpoint_0/currentValue
      transformationPattern: JINJA:{{100-value_json['value']*100/99}}
```

**Comment by apella12** [comment]
I agree auto discovery is so much better, but as noted above, I have uncovered problems with the homeassistant config topics. I think your problem with the position not being reported is related.  From the zwave-js-ui configuration.ts (closely matching what you posted above
`	cover_position: {
		type: 'cover',
		object_id: 'position',
		discovery_payload: {
			state_topic: false,
			command_topic: true,
			position_topic: true,
			set_position_topic: true,
			position_template: '{{ value_json.value | round(0) }}',
			position_open: 99,
			position_closed: 0,
			payload_open: '99',
			payload_close: '0',
		},`
Note that the state_topic is false.

Over on the [Homeassistant](https://www.home-assistant.io/integrations/cover.mqtt) site there is this statement regarding the cover.

`If the cover reports its position, a position_topic can be configured for receiving the position. If no state_topic is configured, the cover’s state will be set to either open or closed when a position is received.`
----
I have been trying to get my head around a way to discover the topics in OH (under the zwave hierarchy) without involving the HA templates.  I have also thought out a hybrid with some HA discovery and a few generic Mqtt things to fill in the blanks (like what you have now), however, some of the configs throw jinjava errors noted in my forum post. Basically just collecting data at this point which I why this issue caught my eye. Hope I helped a bit in understanding at least.
